pvlib/spectrum/mismatch.py

```diff
@@ -571,3 +571,127 @@ def spectral_factor_caballero(precipitable_water, airmass_absolute, aod500,
+
+    .. math::
+
+       M = a_1 K_c^{a_2} AM_a^{a_3},
+
+    where M is the spectral mismatch factor, and :math:`a_1, a_2, a_3` are
+    module-specific coefficients. In the PVSPEC model publication, absolute air
+    mass is estimated using the Kasten and Young model [2]_. The clear sky
+    index, which is the ratio of GHI to clear sky GHI, uses the ESRA model [3]_
+    to estimate the clear sky GHI with monthly Linke turbidity values from [4]_
+    as inputs.
+
+    References
+    ----------
+    .. [1] Pelland, S., Beswick, C., Thevenard, D., Côté, A., Pai, A. and
+       Poissant, Y., 2020, June. Development and testing of the PVSPEC model of
+       photovoltaic spectral mismatch factor. In 2020 47th IEEE Photovoltaic
+       Specialists Conference (PVSC) (pp. 1258-1264). IEEE.
+       :doi:`10.1109/PVSC45281.2020.9300932`
+    .. [2] Kasten, F. and Young, A.T., 1989. Revised optical air mass tables
+       and approximation formula. Applied Optics, 28(22), pp.4735-4738.
+       :doi:`https://doi.org/10.1364/AO.28.004735`
+    .. [3] Rigollier, C., Bauer, O. and Wald, L., 2000. On the clear sky model
+       of the ESRA—European Solar Radiation Atlas—with respect to the Heliosat
+       method. Solar energy, 68(1), pp.33-48.
+       :doi:`https://doi.org/10.1016/S0038-092X(99)00055-9`
+    .. [4] SoDa website monthly Linke turbidity values:
+       `http://www.sodapro.com/gl/web-services/atmosphere/linke-turbidity-
+       factor-ozone-watervapor-and-angstroembeta`_.
+    """
+
+# =============================================================================
+#      --- Default coefficients ---
+# =============================================================================
+    # Empirical coefficients from [1]_
+    _coefficients = {}
+    _coefficients['multisi'] = (
+        0.9847, -0.05237, 0.03034)
+    _coefficients['monosi'] = (
+        0.9845, -0.05169, 0.03034)
+    _coefficients['fs-2'] = (
+        1.002, -0.07108, 0.02465)
+    _coefficients['fs-4'] = (
+        0.9981, -0.05776, 0.02336)
+    _coefficients['cigs'] = (
+        0.9791, -0.03904, 0.03096)
+    _coefficients['asi'] = (
+        1.051, -0.1033, 0.009838)
+    _coefficients['polysi'] = _coefficients['multisi']
+    _coefficients['xsi'] = _coefficients['monosi']
+# =============================================================================
+#      --- Check arguments ---
+# =============================================================================
+    if module_type is not None and coefficients is None:
+        coefficients = _coefficients[module_type.lower()]
+    elif module_type is None and coefficients is not None:
+        pass
+    elif module_type is None and coefficients is None:
+        raise ValueError('No valid input provided, both module_type and ' +
+                         'coefficients are None. module_type can be one of ' +
+                         'poly-Si, monosi, fs-2, fs-4, cigs, or asi')
```

Comment:
```suggestion
                         ", ".join(_coefficients.keys()))
```

This way we don't have to list the keys in more than one place -- it makes it impossible to forget to list one, or forget to update the name here if we change it in the dictionary, etc.

Reply:
Ah this is a very good point

Reply:
Ah this is a good point